client/src/App.jsx

```diff
@@ -1,13 +1,83 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import MapView from './mapView';
 import Interface from './Interface';
+import fetchParklets from '../utils/fetchParklets';
 
-const App = () => (
-  <div id="home">
-    <MapView />
-    <Interface />
-  </div>
-);
+const PARKLETS_ENDPOINT = 'https://data.sfgov.org/resource/6a7x-cttf.json';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      activeParklets: [],
+      hiddenParklets: [],
+      error: '',
+    };
+
+    this.handleFilters = this.handleFilters.bind(this);
+  }
+
+  componentDidMount() {
+    // IMPORTANT: Until db is implemented, fake data will be used to generate flags
+    // for a parklet that is open, has coffee, or wifi by RNG, this will change for
+    // every refresh of the page
+    fetchParklets(PARKLETS_ENDPOINT)
+      .then((data) => {
+        // *******FAKE DATA GENERATOR************
+        const random = () => Math.random() > 0.5;
+
+        data.forEach((d) => {
+          const node = d;
+          node.wifi = random();
+          node.food = random();
+          node.open = random();
+        });
+        // remove above function once graphQL db is configured
+
+        this.setState({ activeParklets: data });
+      })
+      .then(() => this.handleFilters())
+      .catch(err => this.setState({ error: err }));
+  }
+
+  handleFilters() {
+    const open = document.querySelector('.open');
+    const food = document.querySelector('.food');
+    const wifi = document.querySelector('.wifi');
+    const { activeParklets, hiddenParklets } = this.state;
+    const parklets = [...activeParklets, ...hiddenParklets];
+    const show = [];
+    const hide = [];
+
+    parklets.forEach((parklet) => {
+      if (!parklet.open && open.classList.contains('btn-dark-outline')) {
```

Comment:
This whole function is working directly with the DOM. Please restructure to the React way using the virtual DOM.